src/consensus/params.h

```diff
@@ -80,7 +89,19 @@ struct Params {
     int64_t DifficultyAdjustmentInterval() const { return nPowTargetTimespan / nPowTargetSpacing; }
     uint256 nMinimumChainWork;
     uint256 defaultAssumeValid;
+
+    inline int DeploymentHeight(BuriedDeployment dep) const
```

Comment:
Maybe tighter to make this a templated function as we're never calling DeploymentHeight with a variable, only literal.

Then we get not just a warning, but a compiler error.

Reply:
I had tried that, but thought that it might be useful to be able to do `for (dep = 0; dep < MAX_VERSIONBITS_DEPLOYMENTS; ++dep) { ... DeploymentActiveAt(pindex, consParams, dep) .. }`. I think in all the cases where that currently comes up you want to know more fine grained details though. (Changing `BuriedForkDescPushBack` to take a BuriedDeployment instead of a height would make use of that, eg)

I don't think the type checking is any better with the template though, as unfortunately you can invoke `DeploymentHeight<(BuriedDeployment)9000>()` just as easily as `DeploymentHeight((BuriedDeployment)9000)` with both g++ and clang++.